### DISCUSSION TITLE
Don't refresh current location coordinates in the background

### DISCUSSION THREAD

**Discussion by Riajyuu** [breezy-weather/breezy-weather#1314] [Ideas]
### Describe your idea

when updating weather info in background, now the app would still actually request system location API and actually call the hardware. Using cached location wherever available would save battery this case and also avoiding spamming permission request logger on privacy dashboard

### Acknowledgements

- [X] I have searched the [existing issues](https://github.com/breezy-weather/breezy-weather/issues) and [existing discussions](https://github.com/breezy-weather/breezy-weather/discussions/categories/ideas) and this is a new ticket, **NOT** a duplicate or related to another open OR closed issue or discussion.
- [X] The idea I’m suggesting is not in the [list of features that will not be implemented](https://github.com/breezy-weather/breezy-weather/blob/main/README.md#features-that-will-not-be-implemented).
- [X] I have updated the app to **[version 5.2.8 or later](https://github.com/breezy-weather/breezy-weather/releases/latest)**. Even if I’m getting updates from F-Droid default repo, I don’t need to uninstall/reinstall to update.
- [X] I wrote my request in English, and any screenshot I attached are also in English (language can be changed from Settings > Appearance). I can write in my native language under the English text, if necessary.
- [X] I understand that this app is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY, so I may not get a positive reply to my request, if any at all.

**Comment by papjul** [comment]
> when updating weather info in background, now the app would still actually request system location API and actually call the hardware.

We don't call the hardware. We ask Android to use the software, and only in rare cases it will use the hardware (but it's up to Android to decide).

> Using cached location wherever available would save battery this case

What do you mean by "using cached location"? Which one?

If we never request the location in the background, you will never get updated coordinates, unless you also refresh manually in the foreground -which I don't recommend since it means additional calls.

Example: you're located in town A at 14:00, it refreshes. At 15:30 you're far away in town B. It refreshes in the background but with cached location, so with data from town A. If that's your expecting behavior, you probably don't want to use "current location" but add manually town A. Since you were worried about the battery comsumption of the GPS, this would also remove any chance of it being called by Android.

> avoiding spamming permission request

What do you mean? If your refresh rate is -let's say- every 1.5 hours, it will be requested every 1.5 hours. If you experience something else, please share more details.

**Comment by Riajyuu** [comment]
![Screenshot_2024-09-19-08-20-19-662_com miui securitycenter](https://github.com/user-attachments/assets/873fb787-3ab6-4317-a4a7-39246eb8ac7f)
> avoiding spamming permission request

this is what I meant, it spammed privacy dashboard even I set refresh rate at 3hrs

>If we never request the location in the background, you will never get updated coordinates, unless you also refresh manually in the foreground

this is exactly what I asked. not elegant but works

**Reply by papjul** [reply]
> ![Screenshot_2024-09-19-08-20-19-662_com miui securitycenter](https://private-user-images.githubusercontent.com/16912240/368785567-873fb787-3ab6-4317-a4a7-39246eb8ac7f.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY3MTgzMDEsIm5iZiI6MTcyNjcxODAwMSwicGF0aCI6Ii8xNjkxMjI0MC8zNjg3ODU1NjctODczZmI3ODctM2FiNi00MzE3LWE0YTctMzkyNDZlYjhhYzdmLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTE5VDAzNTMyMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdmMDU3MWI3ZmU4YWViOGJiNGI2MTM2MWMyM2RjNWY1NWUzNDI5ZmNhMjk0NjMyZjE0YjcwNmU0ZTdlYjFiYTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.rRmA-U-CcJldz3AYvajEqe35USnkV1Sp3l9VC4sI1aI)

Didn't you refresh in the foreground multiple times? I don't see any reason it would refresh your location every minute.

> > If we never request the location in the background, you will never get updated coordinates, unless you also refresh manually in the foreground
> 
> this is exactly what I asked. not elegant but works

In that case, the solution is simple: turn off "location in the background" permission from Android permission settings of Breezy Weather app.
